Packs/FeedMitreAttackv2/Integrations/FeedMitreAttackv2/FeedMitreAttackv2.py

```diff
@@ -162,11 +167,44 @@ def build_iterator(self, create_relationships=False, is_up_to_6_2=True, limit: i
                                 continue
                             id_to_name[mitre_item_json.get('id')] = value
                             indicator_obj = self.create_indicator(item_type, value, mitre_item_json)
+                            add_obj_to_mitre_id_to_mitre_name(mitre_id_to_mitre_name, mitre_item_json)
                             indicators.append(indicator_obj)
                             counter += 1
                         mitre_id_list.add(mitre_item_json.get('id'))
 
-        return indicators, mitre_relationships_list, id_to_name
+        return indicators, mitre_relationships_list, id_to_name, mitre_id_to_mitre_name
+
+
+def add_obj_to_mitre_id_to_mitre_name(mitre_id_to_mitre_name, mitre_item_json):
+    if mitre_item_json['type'] == 'attack-pattern':
+        mitre_id = [external.get('external_id') for external in mitre_item_json.get('external_references', [])
+                    if external.get('source_name', '') == 'mitre-attack']
+        if mitre_id:
+            mitre_id_to_mitre_name[mitre_id[0]] = mitre_item_json.get('name')
+
+
+def add_technique_prefix_to_sub_technique(indicators, id_to_name, mitre_id_to_mitre_name):
+    for indicator in indicators:
+        if indicator['type'] in ['Attack Pattern', 'STIX Attack Pattern'] and \
+                len(indicator['fields']['mitreid']) > 5:  # Txxxx.xxx is sub technique
+            parent_mitre_id = indicator['fields']['mitreid'][:5]
+            value = indicator['value']
+            technique = mitre_id_to_mitre_name.get(parent_mitre_id)  # TODO in case that technique doesnt exist
+            new_value = f'{technique}: {value}'
+            indicator['value'] = new_value
+            id_to_name[indicator['fields']['stixid']] = new_value
+
+
+def add_malware_prefix_to_dup_with_intrusion_set(indicators):
```

Comment:
maybe it worth explaining what this function is for.
also, note that this pattern of functions that are manipulating the arguments should generally be avoided.
lets talk about it.

Reply:
done